packages/headless/doc-parser/doc-parser.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  ApiItem,
+  ApiItemKind,
+  ApiModel,
+  ApiPropertySignature,
+  ApiEntryPoint,
+  ApiPackage,
+  ApiFunction,
+  Parameter,
+  ExcerptTokenKind,
+  ApiInterface,
+} from '@microsoft/api-extractor-model';
+import {writeFileSync} from 'fs';
+
+const prefix = '@coveo/headless!';
+const apiModel = new ApiModel();
+const apiPackage = apiModel.loadPackage('temp/headless.api.json');
+const entryPoint = findEntryPoint(apiPackage) as ApiEntryPoint;
+
+const config = {
+  initializer: 'buildPager',
+};
+
+const result = getControllerDoc(config.initializer);
+
+writeFileSync('dist/parsed_doc.json', JSON.stringify(result, null, 2));
+
+interface Entity {
+  name: string;
+  type: string;
+  isOptional: boolean;
+  desc: string;
+  // notes?: []
```

Comment:
This property would be to include things like `default`, `min`, `max`, `beta` etc.

Reply:
You mean when we annotate `@default 5` in the JSDoc, it would be "leveraged" by notes ?

How would that be leveraged exactly ?

Reply:
Yes, this script should ultimately include the tags and their values as part of the parsed object and the outputed JSON. From there, the ruby scripts and templates inside the jekyll doc repo would be free to represent the information as they wish, whether in a table or some other structure.